src/components/Editor/index.tsx

```diff
@@ -255,7 +262,8 @@ const Editor = () => {
             return;
         }
         updateWindowUrlHashParams({logEventNum: newLogEventNum});
-    }, []);
+        setLogEventNum(newLogEventNum);
+    }, [setLogEventNum]);
 
     // Synchronize `beginLineNumToLogEventNumRef` with `beginLineNumToLogEventNum`.
```

Comment:
similar to https://github.com/y-scope/yscope-log-viewer/pull/286/files#r2108340456 , we may not no long need to copy states into refs. instead, we can directly read them from the Zustand stores.